subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaBasePlugin.java

```diff
@@ -238,6 +238,8 @@ private void defineConfigurationsForSourceSet(SourceSet sourceSet, Configuration
 
         DeprecatableConfiguration compileOnlyConfiguration = (DeprecatableConfiguration) configurations.maybeCreate(compileOnlyConfigurationName);
         compileOnlyConfiguration.setVisible(false);
+        implementationConfiguration.setCanBeConsumed(false);
```

Comment:
typo? should these two calls be about `compileOnlyConfiguration`?

Reply:
Of course. 🤦  Thanks!